test/840_weave_kube_3_test.sh

```diff
@@ -20,59 +15,56 @@ KUBECTL="sudo kubectl --kubeconfig /etc/kubernetes/admin.conf"
 KUBE_PORT=6443
 IMAGE=weaveworks/network-tester:latest
 
-tear_down_kubeadm
-
-# Make an ipset, so we can check it doesn't get wiped out by Weave Net
-docker_on $HOST1 run --rm --privileged --net=host --entrypoint=/usr/sbin/ipset weaveworks/weave-npc create test_840_ipset bitmap:ip range 192.168.1.0/24 || true
-docker_on $HOST1 run --rm --privileged --net=host --entrypoint=/usr/sbin/ipset weaveworks/weave-npc add test_840_ipset 192.168.1.11
-
-# kubeadm init upgrades to latest Kubernetes version by default, therefore we try to lock the version using the below option:
-k8s_version="$(run_on $HOST1 "kubelet --version" | grep -oP "(?<=Kubernetes )v[\d\.\-beta]+")"
-k8s_version_option="$([[ "$k8s_version" > "v1.6" ]] && echo "kubernetes-version" || echo "use-kubernetes-version")"
-
-for host in $HOSTS; do
-    if [ $host = $HOST1 ] ; then
-	run_on $host "sudo systemctl start kubelet && sudo kubeadm init --$k8s_version_option=$k8s_version --token=$TOKEN"
-    else
-	run_on $host "sudo systemctl start kubelet && sudo kubeadm join --token=$TOKEN $HOST1IP:$KUBE_PORT"
-    fi
-done
-
 if [ -n "$COVERAGE" ]; then
     COVERAGE_ARGS="env:\\n                - name: EXTRA_ARGS\\n                  value: \"-test.coverprofile=/home/weave/cover.prof --\""
 else
     COVERAGE_ARGS="env:"
 fi
 
-# Ensure Kubernetes uses locally built container images and inject code coverage environment variable (or do nothing depending on $COVERAGE):
-sed -e "s%imagePullPolicy: Always%imagePullPolicy: Never%" \
-    -e "s%env:%$COVERAGE_ARGS%" \
-    "$(dirname "$0")/../prog/weave-kube/weave-daemonset-k8s-1.7.yaml" | run_on "$HOST1" "$KUBECTL apply -n kube-system -f -"
+#
+# Utility functions
+#
+tear_down_kubeadm() {
+    for host in $HOSTS; do
+        run_on $host "sudo kubeadm reset && sudo rm -r -f /opt/cni/bin/*weave*"
+    done
+}
+
 
-sleep 5
 
 check_connections() {
     run_on $HOST1 "curl -sS http://127.0.0.1:6784/status | grep \"$SUCCESS\""
 }
 
-assert_raises 'wait_for_x check_connections "connections to establish"'
+check_ready() {
+    run_on $HOST1 "$KUBECTL get nodes | grep -c -w Ready | grep $NUM_HOSTS"
+}
+
+function can_pods_communicate {
+    podName=$($SSH $HOST1 "$KUBECTL get pods -l run=nettest -o go-template='{{(index .items 0).metadata.name}}'")
+    status=$($SSH $HOST1 "$KUBECTL exec $podName -- curl -s -S http://127.0.0.1:8080/status")
+    test "$status" = "pass" && return 0 || return 1
+}
+
+
+#
+# Test functions
+#
+
+function check_all_pods_communicate {
+    assert_raises 'wait_for_x can_pods_communicate pods'
 
-# Check we can ping between the Weave bridg IPs on each host
-HOST1EXPIP=$($SSH $HOST1 "weave expose" || true)
-HOST2EXPIP=$($SSH $HOST2 "weave expose" || true)
-HOST3EXPIP=$($SSH $HOST3 "weave expose" || true)
-assert_raises "run_on $HOST1 $PING $HOST2EXPIP"
-assert_raises "run_on $HOST2 $PING $HOST1EXPIP"
-assert_raises "run_on $HOST3 $PING $HOST2EXPIP"
```

Comment:
Removing the ping tests is rather beyond a "refactor"; if it is intentional please do it in a separate commit and explain.

Reply:
@bboreham Thank you, good catch. I didn't intend to remove those in 840, but in another test.